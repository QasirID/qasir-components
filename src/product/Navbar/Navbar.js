import React from 'react';
import PropTypes from 'prop-types';

import * as Style from './Navbar.styles';
import ArrowBack from '../../base/svg-icon/svg/ArrowBack';

class NavBar extends React.Component {
  eventOnClick = item => {
    this.props.onClick && this.props.onClick(item);
  };

  render() {
    const {
      title,
      imageSrc,
      className,
      useLeftIcon,
      rightContent,
      ...other
    } = this.props;

    return (
      <Style.Wrapper className={`qasir-navbar ${className}`} {...other}>
        <Style.Container className="container">
          {useLeftIcon && (
            <Style.Icon onClick={this.eventOnClick}>
              {imageSrc ? (
                <img src={imageSrc} alt={title} />
              ) : (
                <ArrowBack width={20} height={20} fill="#F04B32" />
              )}
            </Style.Icon>
          )}
          <Style.Title>{title}</Style.Title>
        </Style.Container>
        {rightContent && (
          <Style.RightContent> {rightContent} </Style.RightContent>
        )}
      </Style.Wrapper>
    );
  }
}

NavBar.propTypes = {
  /**
   * Class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Title of page
   */
  title: PropTypes.string,
  /**
   * Image Icon of Navbar
   */
  imageSrc: PropTypes.string,
  /**
   * Show / hide left Icon
   */
  useLeftIcon: PropTypes.bool,
  onClick: PropTypes.func,
  /**
   * Add right content element if any, you can use your own component
   */
  rightContent: PropTypes.node
};

NavBar.defaultProps = {
  useLeftIcon: true,
  className: ''
};

export default NavBar;
