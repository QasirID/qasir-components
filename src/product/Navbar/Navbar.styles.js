import styled from 'styled-components';
import * as c from '../../shared/constants/color';
import * as v from '../../shared/constants/variables';

export const Wrapper = styled.div`
  height: 56px;
  padding: 0 20px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: fixed;
  flex-direction: row;
  width: 100%;
  z-index: 987;
  top: 0;
  background-color: white;
  box-shadow: 0px 4px 12px rgba(0, 0, 0, 0.04);
  left: 0;

  &.bg-transparent {
    background-color: transparent;
  }
`;

export const Container = styled.div`
  display: flex;
  align-items: center;
  flex-direction: row;
  width: calc(100% - 100px);
`;

export const Title = styled.div`
  color: ${c.DarkPrimary};
  font-size: 16px;
  font-weight: 600;
  width: 100%;
  position: relative;
  line-height: 1;
  text-transform: capitalize;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const Icon = styled.div`
  width: 20px;
  cursor: pointer;
  margin-right: ${v.spacing.ml}px;
  img,
  svg {
    max-width: 100%;
  }
`;

export const RightContent = styled.div`
  margin-left: ${v.spacing.ml}px;
`;
