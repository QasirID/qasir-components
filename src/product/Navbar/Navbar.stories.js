/* eslint-disable no-alert */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';

import Navbar from './Navbar';
import Report from '../../base/svg-icon/svg/Report';
import Notification from '../../base/svg-icon/svg/Notification';

const Additional = (
  <div style={{ display: 'flex', flexDirection: 'row' }}>
    <div style={{ marginRight: 15 }}>
      <Report />
    </div>
    <div>
      <Notification />
    </div>
  </div>
);

/**
 * Put here:
 * - description
 */
const DESCRIPTION = `
Navbar
---
~~~js
import Navbar from 'qasir/lib/product';
~~~

or

~~~js
import Navbar from 'qasir/lib/product/Navbar';
~~~

Examples
~~~js
<Navbar {...props} />
~~~
`;

storiesOf('product|Navbar', module)
  .addParameters({
    info: {
      text: DESCRIPTION
    }
  })
  .add('preview with knobs', () => (
    <Navbar
      imageSrc={text('imageSrc', '')}
      className={text('className', '')}
      title={text('title', 'Profile')}
      useLeftIcon={boolean('useLeftIcon', true)}
      onClick={() => alert('triggered')}
    />
  ))
  .add('with right content', () => (
    <Navbar
      title={'Profile'}
      onClick={() => alert('triggered')}
      rightContent={Additional}
    />
  ));
