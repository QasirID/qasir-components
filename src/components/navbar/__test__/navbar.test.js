import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import Navbar from '..';

describe('Navbar component', () => {
    it('Should exist', () => {
        expect(Navbar).toMatchSnapshot();
    });
    it('Should be a  Navbar', () => {
        const component = renderer.create(<Navbar />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});