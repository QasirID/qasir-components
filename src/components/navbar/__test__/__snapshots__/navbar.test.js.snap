// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should be a default Button 1`] = `
<Navbar
  onClick='' 
  title={'Deposit'}
  iconRight={true} 
  imgRight=''
  badge={0}
  badgeCount={1}
  toLink=''>
</Navbar>
`;

exports[`Button component Should exist 1`] = `[Function]`;
